RELEASE/scripts/autoscend/iotms/mr2021.ash

```diff
@@ -112,3 +112,245 @@ int auto_backupUsesLeft()
+		}
+		// Single step.
+		else if (available_amount($item[battery (AA)]) >= 2)
+		{
+			craft("combine", 1, $item[battery (AA)], $item[battery (AA)]);
+			return (available_amount($item[battery (9-Volt)]) >= 1);
+		}
+		// Every multi step case with recusion.
+		else if (available_amount($item[battery (AAA)]) >= 4 ||
+		 (available_amount($item[battery (AA)]) >= 1 && available_amount($item[battery (AAA)]) >= 2))
+		{
+			batteryCombine($item[battery (D)]);
+			craft("combine", 1, $item[battery (D)], $item[battery (AAA)]);
+			return (available_amount($item[battery (9-Volt)]) >= 1);
+		}
+		return false;
+	}
+
+	else if (battery == $item[battery (lantern)])
+	{
+		// Single step.
+		if (available_amount($item[battery (9-Volt)]) >= 1 && available_amount($item[battery (AAA)]) >= 1)
+		{
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AAA)]);
+			return (available_amount($item[battery (lantern)]) >= 1);
+		}
+		// Single step.
+		else if (available_amount($item[battery (D)]) >= 1 && available_amount($item[battery (AA)]) >= 1)
+		{
+			craft("combine", 1, $item[battery (D)], $item[battery (AA)]);
+			return (available_amount($item[battery (lantern)]) >= 1);
+		}
+		// Every multi step case with recusion.
+		else if (available_amount($item[battery (AAA)]) >= 5 ||
+		 (available_amount($item[battery (AA)]) >= 1 && available_amount($item[battery (AAA)]) >= 3) ||
+		 (available_amount($item[battery (D)]) >= 1 && available_amount($item[battery (AAA)]) >= 2) ||
+		 (available_amount($item[battery (AA)]) >= 2 && available_amount($item[battery (AAA)]) >= 1))
+		{
+			batteryCombine($item[battery (9-Volt)]);
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AAA)]);
+			return (available_amount($item[battery (lantern)]) >= 1);
+		}
+		return false;
+	}
+
+	else if (battery == $item[battery (car)])
+	{
+		// Single step.
+		if (available_amount($item[battery (lantern)]) >= 1 && available_amount($item[battery (AAA)]) >= 1)
+		{
+			craft("combine", 1, $item[battery (lantern)], $item[battery (AAA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Single step.
+		else if (available_amount($item[battery (9-Volt)]) >= 1 && available_amount($item[battery (AA)]) >= 1)
+		{
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Single step.
+		else if (available_amount($item[battery (D)]) >= 2)
+		{
+			craft("combine", 1, $item[battery (D)], $item[battery (D)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// The only multi-step case that can't be resolved by the same function (can't turn AAs into a lantern without a AA or D)
+		else if (available_amount($item[battery (AA)]) >= 3)
+		{
+			batteryCombine($item[battery (9-volt)]);
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Every other multi step case with recusion.

```

Comment:
very minor spelling issue.
you used the word recusion in the comment 3 times and it is recursion with an r between the u and s

Reply:
copy and paste is a bear, fixing right now.

Reply:
I can see it is fixed now. resolving this convo